src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
 
+            gSPEndDisplayList(sp74++);
+            Graph_BranchDlist(sp70, sp74);
+            POLY_OPA_DISP = sp74;
+            this->unk_18B49 = 2;
+            SREG(33) |= 1;
+        } else {
+        Play_Draw_PostWorldDraw:
+            Play_PostWorldDraw(this);
+        }
+    }
+
+    //! Fake: regalloc around here
+    if (this && this && this) {}
+
+    if ((this->view.unk164 != 0) && !gDbgCamEnabled) {
+        Vec3s sp4C;
+
+        Camera_Update(&sp4C, GET_ACTIVE_CAM(this));
+        View_UpdateViewingMatrix(&this->view);
+        this->view.unk164 = 0;
+        if ((this->skyboxId != 0) && !this->envCtx.skyboxDisabled) {
+            SkyboxDraw_UpdateMatrix(&this->skyboxCtx, this->view.eye.x, this->view.eye.y, this->view.eye.z);
+        }
+    }
+
+    if (!sp25B) {
+        func_800FE3E0(this);
+    }
+
+    CLOSE_DISPS(this->state.gfxCtx);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Play_Draw.s")
+#endif
+
+void Play_SetupDraw(PlayState* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+
+    {
+        GraphicsContext* gfxCtx2 = this->state.gfxCtx;
+
+        if (sBombersNotebookOpen) {
+            if (D_801FBBD4 != 1) {
+                MsgEvent_SendNullTask();
+                func_80178818();
+                gfxCtx2->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx2->framebufferIdx % 2);
+                gfxCtx2->zbuffer = SysCfb_GetZBuffer();
+                gfxCtx2->viMode = D_801FBB88;
+                gfxCtx2->viConfigFeatures = gViConfigFeatures;
+                gfxCtx2->xScale = gViConfigXScale;
+                gfxCtx2->yScale = gViConfigYScale;
+                gfxCtx2->updateViMode = true;
+            }
+        } else {
+            if (D_801FBBD4 != 0) {
+                MsgEvent_SendNullTask();
+                func_80178750();
+                gfxCtx2->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx2->framebufferIdx % 2);
+                gfxCtx2->zbuffer = SysCfb_GetZBuffer();
+                gfxCtx2->viMode = D_801FBB88;
+                gfxCtx2->viConfigFeatures = gViConfigFeatures;
+                gfxCtx2->xScale = gViConfigXScale;
+                gfxCtx2->yScale = gViConfigYScale;
+                gfxCtx2->updateViMode = true;
+            }
+        }
+    }
+
+    if (sBombersNotebookOpen && ((SREG(2) != 2) || (gZBufferPtr == NULL))) {
+        Hireso_Draw(&sHireso, gfxCtx);
+        Message_Draw(this);
+    } else {
+        Play_Draw(this);
+    }
+}
+
+void Play_Main(GameState* thisx) {
+    static Input* prevInput = NULL;
+    PlayState* this = (PlayState*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80168DAC.s")
+    prevInput = CONTROLLER1(&this->state);
+    DebugDisplay_Init();
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Play_Main.s")
+    {
+        GraphicsContext* gfxCtx = this->state.gfxCtx;
+
+        if (1) {
+            this->state.gfxCtx = NULL;
+        }
+        Play_SetupUpdate(this);
+        this->state.gfxCtx = gfxCtx;
```

Comment:
these "setup" functions for update and draw seem to really be about handling the bombers notebook while its open, and if its not it just passes it onto the normal update and draw functions. Im wondering if we can name it more based on that, instead of implying that its setting things up for the main update and draw processes

part of me wants these "setup" functions to actually be named update and draw, because they are the direct functions called by main. but I can understand why that may not be desirable

Reply:
I'm leaning more and more this way, does anyone have any objections?

Reply:
Fig suggested `Play_UpdateGame/Play_DrawGame` in the discord, which I agree with.

Reply:
it was more of an example name, Im not 100% for it cause it implies that the other gamestates arent the game as a whole, which they are. so its kinda weird. 
And one may argue that "game" has closer meaning to gamestate cause of its original name, but I think we are passed that point by now 

"gameplay" is also not great imo, but at least in oot we do have things using that term like `Player_DrawGameplay` `Player_OverrideLimbDrawGameplayDefault` `Player_OverrideLimbDrawGameplayFirstPerson`
etc, which is conveying the same meaning of "gameplay" as we are trying to here

Reply:
It's a fair question to what extent we want to sync those up, since for the Player they're explicitly "when kaleido isn't open", which still counts as the non–Bombers Notebook draw here; they're similar, but not quite the same.

Reply:
even though the non-gameplay state being referred to here is different, the same things are meant by gameplay. so I think it makes sense. But yea idk if this terminology is going to stick around long term

Reply:
Still considering different names, but went ahead and implemented this using just `UpdateGame` and `DrawGame` for now.

Reply:
I don't want `Game` for the same reason we decided not to use it for gamestates: the whole thing is the game, so it doesn't really make sense to use it to describe a subset.